### PR TITLE
 Add small QoL improvements to repl 

### PR DESCRIPTION
**Pull request by jam1garner**
* Highlight shell prompt to make commands more readable
* eval with no arguments passed will switch to a js shell where "eval "
is assumed
* Hosting IP address is output in the startup message instead of only outside of
repl
* Whitespace in from of commands will be ignored

### CODE REVIEW

**Comment by SomeoneWeird** [comment]
Nice. I just pushed a commit that added an eslint configuration file that it seems was missed when we open sourced the repo, can you please make sure your code conforms to that? 👍

**Comment by SomeoneWeird** [comment]
Also, can you update the readme so it explains the mechanics of the new eval functionality?

**Comment by jam1garner** [comment]
Yeah sure no problem

**Comment by misson20000** [comment]
Not pleased with the way you changed the message that shows the IP. First of all, that's not the ip it listens on, it's the one it sticks in responses (my machine is on like 4 different networks so I usually have to manually specify it and that helps me make sure I'm on the right one). The one it listens on is host. `ipAddr` is now defined twice, and I'm not a fan of that code duplication

**Comment by jam1garner** [comment]
Added the requested changes and @misson20000 could you check behind me on the hosting part? I'm not certain I did it right because I have same ip for both.

File: repl.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ const fns = {
 	eval: {
 		response: 'evald',
 		help: 'eval <...code>',
-		helptxt: 'Evals code on remote console and returns response if applicable'
+		helptxt: 'Evals code on remote console and returns response if applicable, if no arguments are passed it will switch to js shell instead'
```
Comment (misson20000) [inline repl.js]: `applicable, if` -> `applicable. If`

File: start.js [hunk 2]
```diff
@@ -329,8 +329,7 @@ Promise.all([dnsServerStarted, httpServerStarted]).then(() => {
 
 		// Render everything
 		screen.render();
-		let ipAddr = argv.ip || ip.address();
-		repl.write("Waiting for DNS connection at ip " + ipAddr + "...");
+		repl.write("Waiting for DNS connection at ip " + (argv.host || ip.address()) + "...");
```
Comment (misson20000) [inline start.js]: We're waiting for dns requests (not connections) on `argv.host`, but the important part is that we're responding with the ip in `ipAddr`. `ipAddr` had already been defined on line 62. I don't know if it's still in scope here or not, but if it's not, you can probably move the declaration to bring it in scope here.

**Comment by jam1garner** [comment]
Made requested changes

File: start.js [hunk 3]
```diff
@@ -329,6 +329,8 @@ Promise.all([dnsServerStarted, httpServerStarted]).then(() => {
 
 		// Render everything
 		screen.render();
+		repl.write("Responding with address " + ipAddr + "\r\n");
```
Comment (misson20000) [inline start.js]: This seems to only get printed if curses is enabled

**Comment by misson20000** [comment]
```
$ sudo node start.js --disable-curses --logfile log.txt --ip 10.0.0.192 --setuid misson20000
DNS Server started, responding with address 10.0.0.192

switch> 
Switch connected...
switch> eval

utils.log("foo")
'no output'
switch/js> utils.log("bar")
'no output'
switch/js> 


switch> 

```
Looks like something weird happens when you change the prompt. This happens both when curses is enabled and when it's not. You might also want to consider printing a message saying how to get out of `switch/js>` mode when you enter it. Aside from the issue with the prompt and that, I like it and once those are fixed I think it'll be ready to merge.

**Pull request merged**